src/engine/cancun.md

```diff
@@ -170,3 +170,26 @@ This document introduces deprecation of [`engine_exchangeTransitionConfiguration
 2. Execution layer clients **MUST NOT** surface an error message to the user if this method is not called.
 
 3. Consensus and execution layer clients **MAY** remove support of this method after Cancun. If no longer supported, this method **MUST** be removed from the [`engine_exchangeCapabilities`](./common.md#engine_exchangecapabilities) request or response list depending on whether it is consensus or execution layer client.
+
+
+### Update methods of previous forks
+
+For the following methods:
+
+- `engine_newPayloadV1`
```

Comment:
`V1` methods must not work for Shanghai as well. I see two options, 1) be strict and say that they must return Unsupported fork for payloads starting from Shanghai 2) leave them unchanged as we unlikely face issues related to `V1` misuse considering a switch from `V2` to `V3`

Reply:
What if we go further and make all the methods decline any timestamp outside of time frame of their exact fork? Seems simpler and more consistent.

By the way, it might be better if every method did not need to know anything about previous or future versions of it. For example, engine_newPayloadV1 detects V2 sub-fields and declines payload if such sub-fields present. Instead we could say all methods require exact set of fields (or instead all methods can ignore additional fields).

Reply:
I am all for being more strict as you describing. I just don't want to overload the spec with things clients won't implement. This is why I think Cancun spec should omit any changes to `V1`, and only touch `V2` as this can be really helpful in discovering bugs at early stages.

The structure check should be extended with something like the following:

2. Client software **MUST** check that provided set of parameters and their fields strictly matches the expected one and return `-32602: Invalid params` error if this check fails. Any field having `null` value **MUST** be considered as not provided.